src/main/scala/com/databricks/spark/redshift/AWSCredentialsUtils.scala

```diff
@@ -71,8 +69,8 @@ private[redshift] object AWSCredentialsUtils {
           }
         }.orElse {
           // Next, try to read from configuration
-          val accessKey = hadoopConfiguration.get("fs.s3a.access.key", null)
-          val secretKey = hadoopConfiguration.get("fs.s3a.secret.key", null)
+          val accessKey = hadoopConfiguration.get(s"fs.$uriScheme.access.key", null)
```

Comment:
As far as I know `fs.s3n.access.key` won't work; S3A uses a different scheme for encoding credentials than the other file systems.